moneyflow/backends/ynab.py

```diff
@@ -0,0 +1,161 @@
+"""
+YNAB backend implementation.
+
+Wraps the YNABClient to implement the FinanceBackend interface.
+"""
+
+from typing import Any, Dict, List, Optional
+
+from ..ynab_client import YNABClient
+from .base import FinanceBackend
+
+
+class YNABBackend(FinanceBackend):
+    """
+    YNAB backend implementation.
+
+    This wraps the YNABClient to provide a standardized
+    FinanceBackend interface for moneyflow.
+    """
+
+    def __init__(self):
+        """Initialize the YNAB backend."""
+        self.client = YNABClient()
+
+    async def login(
+        self,
+        email: Optional[str] = None,
+        password: Optional[str] = None,
+        use_saved_session: bool = True,
+        save_session: bool = True,
+        mfa_secret_key: Optional[str] = None,
+    ) -> None:
+        """
+        Authenticate with YNAB using a Personal Access Token.
+
+        Args:
+            email: Not used (YNAB uses token auth)
+            password: YNAB Personal Access Token
+            use_saved_session: Not used (token is stateless)
+            save_session: Not used (token is stateless)
+            mfa_secret_key: Not used (YNAB doesn't use MFA)
+
+        Raises:
+            ValueError: If no token provided or no budgets found
+        """
+        if not password:
+            raise ValueError(
+                "YNAB backend requires an access token. "
+                "The access token should be stored in the password field."
+            )
+        self.client.login(password)
+
+    async def get_transactions(
+        self,
+        limit: int = 100,
+        offset: int = 0,
+        start_date: Optional[str] = None,
+        end_date: Optional[str] = None,
+        **kwargs,
+    ) -> Dict[str, Any]:
+        """
+        Fetch transactions from YNAB.
+
+        Args:
+            limit: Maximum number of transactions to return
+            offset: Number of transactions to skip (for pagination)
+            start_date: Filter transactions from this date (ISO format: YYYY-MM-DD)
+            end_date: Not supported by YNAB API
+            **kwargs: Additional parameters (e.g., hidden_from_reports)
+
+        Returns:
+            Dictionary containing transaction data in Monarch-compatible format
+        """
+        return self.client.get_transactions(
+            limit=limit,
+            offset=offset,
+            start_date=start_date,
+            **kwargs,
+        )
+
+    async def get_transaction_categories(self) -> Dict[str, Any]:
+        """
+        Fetch all available transaction categories from YNAB.
+
+        Returns:
+            Dictionary containing categories in Monarch-compatible format
```

Comment:
nit: let's call this "moneyflow-compatible". The category normalization / mapping is a bit tedious since every system has slightly different categories. We probably will need to provide some CLI options to generate a default category file for `~/.moneyflow/config.yaml` to align the behavior to a YNAB user, or a Monarch user, for example (rather than foisting Monarch-centric categories)